### PR TITLE
UX: Multichain: Improve accessibility of accounts menu

### PR DESCRIPTION
**Pull request by darkwing**
## Explanation

The original implementation of the `AccountListItem` was kinda hacky; it was an `<a>` that contains a `[button as span]` and was kinda weird.  After speaking with @georgewrmarshall and looking at what GitHub does, I think this solution is much nicer.

## Screenshots/Screencaps

The account menu items look exactly the same; instead, when tabbing, you'll stop on just the account name instead of the whole block.

## Pre-merge author checklist

- [ ] I've clearly explained:
  - [ ] What problem this PR is solving
  - [ ] How this problem was solved
  - [ ] How reviewers can test my changes
- [ ] Sufficient automated test coverage has been added

## Pre-merge reviewer checklist

- [ ] Manual testing (e.g. pull and build branch, run in browser, test code being changed)
- [ ] PR is linked to the appropriate GitHub issue
- [ ] **IF** this PR fixes a bug in the release milestone, add this PR to the release milestone

If further QA is required (e.g. new feature, complex testing steps, large refactor), add the `Extension QA Board` label.

In this case, a QA Engineer approval will be be required.

### CODE REVIEW

File: ui/components/multichain/account-list-item/account-list-item.js [hunk 1]
```diff
@@ -127,10 +125,14 @@ export const AccountListItem = ({
                   position="bottom"
                   wrapperClassName="multichain-account-list-item__tooltip"
                 >
-                  {identity.name}
+                  <a href="#" onClick={onClick}>
+                    {identity.name}
+                  </a>
```
Comment (NidhiKJha) [inline ui/components/multichain/account-list-item/account-list-item.js:130]: Can we use ButtonLink here instead of `a`?

Comment (vthomas13) [reply ui/components/multichain/account-list-item/account-list-item.js:130]: I'm curious about whether ButtonLink is good here too

Comment (georgewrmarshall) [reply ui/components/multichain/account-list-item/account-list-item.js]: I think the intention here is to keep the UI looking like static text but allow for keyboard navigation and tabbing. I guess the difference to this and GitHub is that they're using it for navigation while here we are using an onClick to change the state of the app not necessarily the route

File: ui/components/multichain/account-list-item/account-list-item.js [hunk 2]
```diff
@@ -127,10 +126,20 @@ export const AccountListItem = ({
                   position="bottom"
                   wrapperClassName="multichain-account-list-item__tooltip"
                 >
-                  {identity.name}
+                  <ButtonLink
+                    onClick={onClick}
+                    className="multichain-account-list-item__account-name"
+                  >
+                    {identity.name}
+                  </ButtonLink>
```
Comment (georgewrmarshall) [inline ui/components/multichain/account-list-item/account-list-item.js:134]: I think `<Text as="button">` we want to keep the Text styling but change the underlying element. I think you will still need to add some custom styles here via Box props or a className for the hover states though

**Review by georgewrmarshall — COMMENTED** [review]
Left one suggestion. Keyboard navigation works great though!! I think a `button` element is the right move semantically 🦾

File: ui/components/multichain/account-list-item/account-list-item.js (hunk 2 above)
Comment (darkwing) [reply ui/components/multichain/account-list-item/account-list-item.js]: That isn't working as it bricks the `ellipsis` functionality.  Also, using `ButtonLink` requires only one CSS change, while using `as="button"` requires some background and padding work.

**Review by brad-decker — DISMISSED** [review]

File: ui/components/multichain/account-list-item/account-list-item.js [hunk 3]
```diff
@@ -121,17 +120,24 @@ export const AccountListItem = ({
...
-              ) : (
-                identity.name
-              )}
+              <ButtonLink
+                onClick={onClick}
+                className="multichain-account-list-item__account-name"
+                color={Color.textDefault}
+                ellipsis
+              >
+                {identity.name.length > MAXIMUM_CHARACTERS_WITHOUT_TOOLTIP ? (
```
Comment (vthomas13) [inline ui/components/multichain/account-list-item/account-list-item.js:129]: I feel like this logic can be reused in the future, really nice

**Review by vthomas13 — APPROVED** [review]

**Review by NidhiKJha — APPROVED** [review]
LGTM! Just a non blocking suggestion

**Pull request merged**